src/luaclang.c

```diff
@@ -1,14 +1,312 @@
 #include <clang-c/Index.h>
+#include <stdbool.h>
+
 #include "lua.h"
 #include "lualib.h"
 #include "lauxlib.h"
 
-static const struct luaL_Reg luaclang [] = {
-      {NULL, NULL}  /* sentinel */
+#define CLANG_METATABLE  "Clang.Object"
+#define CURSOR_METATABLE "Clang.Cursor"
+#define TYPE_METATABLE   "Clang.Type"
+
+typedef struct clang_object {
+        CXIndex idx;
+        CXTranslationUnit tu;
+} clang_object;
+
+/* Create clang_object as userdata */
+static clang_object *new_clangobject(lua_State *L) 
+{
+        clang_object *clang_parser = (clang_object*) lua_newuserdata(L, sizeof(clang_object));
+        luaL_getmetatable(L, CLANG_METATABLE);
+        lua_setmetatable(L, -2);
+        return clang_parser;
+}
+
+/* Convert userdata type to  */
+static clang_object to_clangobject(lua_State *L, int n) 
```

Comment:
You should return a pointer, else you won't be able to assign NULL to its fields later.
Probably, this is the reason that you are getting a segfault.